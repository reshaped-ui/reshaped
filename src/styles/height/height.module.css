/* --rs-h is used in other components */
@responsive [style*="--rs-h"] {
	@variable --rs-h auto;
}

@responsive .- {
	@value literal {
		height: var(--rs-h) !important;
	}

	@value unit {
		height: calc(var(--rs-h) * var(--rs-unit-x1)) !important;
	}
}
