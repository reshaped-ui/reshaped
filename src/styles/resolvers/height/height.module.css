/* 
--rs-h is used in other components
we're resolving --rs-h separately to make sure it uses the resolved value in other components calc (e.g. Avatar)
*/
.root {
	--rs-h-s: auto;
	--rs-h-m: var(--rs-h-s);
	--rs-h-l: var(--rs-h-m);
	--rs-h-xl: var(--rs-h-l);

	height: var(--rs-h) !important;
}

.--type-literal {
	--rs-h: var(--rs-h-s);
}

.--type-unit {
	--rs-h: calc(var(--rs-h-s) * var(--rs-unit-x1));
}

@each $viewport in m, l, xl {
	@media (--rs-viewport-$viewport) {
		.--type-literal--$viewport {
			--rs-h: var(--rs-h-$viewport);
		}

		.--type-unit--$viewport {
			--rs-h: calc(var(--rs-h-$viewport) * var(--rs-unit-x1));
		}
	}
}
