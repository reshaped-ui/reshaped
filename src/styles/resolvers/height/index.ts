import { responsiveClassNames, responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import s from "./height.module.css";

const getHeightStyles: T.StyleResolver<T.Height> = (value) => {
	if (!value) return {};
	const variables = responsiveVariables("--rs-h", value);
	const classNames = responsiveClassNames(
		s,
		(value) => (typeof value === "number" ? "--type-unit" : "--type-literal"),
		value,
		{ excludeValueFromClassName: true }
	);

	return { classNames: [s.root, classNames], variables };
};

export default getHeightStyles;
