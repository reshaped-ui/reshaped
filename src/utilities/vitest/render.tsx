import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
