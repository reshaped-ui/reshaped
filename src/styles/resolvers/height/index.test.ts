import { expect, test, describe } from "vitest";
import height from "./index";

describe("Styles/Height", () => {
	test("handles px value", () => {
		expect(height("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(height("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(height(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(height()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(height({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
