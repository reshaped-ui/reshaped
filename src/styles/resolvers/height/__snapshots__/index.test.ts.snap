// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Height > handles % value 1`] = `
{
  "classNames": [
    "_root_4e7a72",
    [
      "_--type-literal_4e7a72",
    ],
  ],
  "variables": {
    "--rs-h-s": "50%",
  },
}
`;

exports[`Styles/Height > handles px value 1`] = `
{
  "classNames": [
    "_root_4e7a72",
    [
      "_--type-literal_4e7a72",
    ],
  ],
  "variables": {
    "--rs-h-s": "50px",
  },
}
`;

exports[`Styles/Height > handles responsive value 1`] = `
{
  "classNames": [
    "_root_4e7a72",
    [
      "_--type-literal_4e7a72",
      "_--type-literal--l_4e7a72",
    ],
  ],
  "variables": {
    "--rs-h-l": "50%",
    "--rs-h-s": "50px",
  },
}
`;

exports[`Styles/Height > handles undefined value 1`] = `{}`;

exports[`Styles/Height > handles unit value 1`] = `
{
  "classNames": [
    "_root_4e7a72",
    [
      "_--type-unit_4e7a72",
    ],
  ],
  "variables": {
    "--rs-h-s": 5,
  },
}
`;
